### PR TITLE
add support for ember-data-model-fragments

### LINKED ISSUE #23: Model with fragment error
Hi, I have an 2.14 ember-data model that includes a fragment.

After updating ember-data-copyable (which now internally uses ember-copy) I get "Fatal error: TypeError: Cannot read property 'serialize' of undefined" errors. 

I narrowed it down to a check that is done to see if the fragment has the copyable mixin. It returns false while true is expected. The Fragment has the Copyable Mixin.

https://github.com/offirgolan/ember-data-copyable/blob/966e1cb534104d0bcd32eba2625c46d7573faba6/addon/mixins/copyable.js#L147 

I don't understand where the detect method comes from. Can't find it here https://github.com/emberjs/ember-copy/search?q=detect

For now I downgraded...

### PR DESCRIPTION
**Pull request by jakesjews**
Fix copying model fragments. Resolves #23

### CODE REVIEW

File: addon/mixins/copyable.js [hunk 1]
```diff
@@ -154,14 +154,14 @@ export default Mixin.create({
       ) {
         let value = this.get(name);
 
-        if (Copyable && Copyable.detect(value)) {
+        if ((Copyable && Copyable.detect(value)) || (value && value._isFragment)) {
```
Comment (offirgolan) [inline addon/mixins/copyable.js]: Are all model fragments copyable?

File: addon/mixins/copyable.js [hunk 2]
```diff
@@ -154,14 +154,14 @@ export default Mixin.create({
...
           // "value" is an Ember.Object using the ember-copy addon
           // (ie. old deprecated Ember.Copyable API - if you use
           // the "Ember Data Model Fragments" addon and "value" is a fragment or
           // if use your own serializer where you deserialize a value to an
           // Ember.Object using this Ember.Copyable API)
           value = value.copy(deep);
-        } else {
+        } else if (value) {
```
Comment (offirgolan) [inline addon/mixins/copyable.js]: What is the reason for this change? Previously all values should be run through their transforms even if falsy. For example, you can have a vale of `NaN` that can have a custom transform to be `Not a Number` or `0`, etc...

File: addon/mixins/copyable.js (hunk 1 above)
Comment (jakesjews) [reply addon/mixins/copyable.js]: They are. For some reason Copyable.detect didn't seem to work.

File: addon/mixins/copyable.js (hunk 2 above)
Comment (jakesjews) [reply addon/mixins/copyable.js]: It's to prevent serializing a null fragment since that throws an exception. I can change the check to make sure its not a fragment to prevent reaching this branch in that case.

Comment (offirgolan) [reply addon/mixins/copyable.js]: > I can change the check to make sure its not a fragment to prevent reaching this branch in that case.

Yeah, I would rather the check be explicit.

**Comment by jakesjews** [comment]
Looks like the CI failure is from a new ember-data. I'm not quite sure if its from my changes

**Comment by offirgolan** [comment]
All good, Just beta and canary are failing. Changes look good. Thanks for taking this up @jakesjews!

**Comment by offirgolan** [comment]
Released with `v1.1.0`.

**Comment by jakesjews** [comment]
thanks!
